federation-js/src/types.ts

```diff
@@ -87,6 +98,15 @@ export const entitiesField: GraphQLFieldConfig<any, any> = {
         );
       }
 
+      if (info.cacheControl && info.cacheControl.cacheHint.replace) {
+        const cacheHint: CacheHint | undefined =
+          info.cacheControl.cacheHintFromType(type);
+
+        if (cacheHint) {
+          info.cacheControl.cacheHint.replace(cacheHint);
```

Comment:
This should be restrict, but requires the bug fix mentioned above.

Reply:
Fix released in 3.0.1 so it should be able to be restrict. I think it's fine for us to release this with the caveat that it doesn't work with 3.0.0.

Reply:
Updated to use the `restrict` method with 3.0.1